src/doc_futureplans.rst

```diff
@@ -0,0 +1,390 @@
+.. _doc_futureplans:
+
+============
+Future Plans
+============
+
+This section contains a description of upcoming high-level changes to the VarFish software.
+The sections below have been extrapolated from the current issue list.
+
+
+.. _doc_futureplans_technical_debt:
+
+--------------
+Technical Debt
+--------------
+
+There is some technical debt, some notable items.
+
+Automated Tests
+    We need more automated tests in various areas throughout the codebase.
+
+Python Type Annotations
+    We should modernize the codebase by comprehensive use of Python type annotations.
+    Notable, this will require type annotations for sodar-core.
```

Comment:
Do we already have mypy in our CI? Otherwise maybe mention that.

Reply:
IMO that goes without saying. Actually, pyright is the cool new engine to use ;-)